src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+
+  width: $width-size;
+  padding: 20px $padding-size;
+  gap: $gap-size;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    @include square($square-size);
+    align-items: center;
+
+    border: 1px solid black;
+    background-color: #eee;
+
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+
+    cursor: pointer;
+
+    &:hover {
+      background-color: pink;
+
+      transform: translateY(-20px);
+      transition: 0.5s;
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(n + #{$i}):before {
+        content: "#{$i}";
+        position: absolute;
+      }
+    }
+
+    @each $day, $week-day in $week-days {
+      &--start-day--#{$day}
+      &__day:first-of-type {
+        margin-left: $week-day;
+      };
+    }
```

Comment:
this @each doesn't work(( I couldn't change your calendar to another first day(( Maybe in margin-left, you should set some function? Did you see what in your css files? In this case $week-day - it's a 1,2,3,4,5,6,7 as I understood.